index.php

```diff
@@ -362,55 +254,15 @@ function get_type_affinity($type)
 }
 
 
-// Returns a file size limit in bytes based on the PHP upload_max_filesize
-// post_max_size and memory_limit. Returns -1 in case of no limit.
-function fileUploadMaxSize()
-{
-	$max1 = parseSize(ini_get('post_max_size'));
-	$max2 = parseSize(ini_get('upload_max_filesize'));
-	$max3 = parseSize(ini_get('memory_limit'));
-	if($max1>0 && ($max1<=$max2 || $max2==0) && ($max1<=$max3 || $max3==-1))
-		return $max1;
-	elseif($max2>0 && ($max2<=$max1 || $max1==0) && ($max2<=$max3 || $max3==-1))
-		return $max2;
-	elseif($max3>-1 && ($max3<=$max1 || $max1==0) && ($max3<=$max2 || $max2==0))
-		return $max3;
-	else
-		return -1; // no limit
-}
-
-// Parses given size string like "12M" into number of bytes
-// based on https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Bytes.php/function/Bytes%3A%3AtoInt/8.2.x
-function parseSize($size)
-{
-	// Remove the non-unit characters from the size.
-	$unit = preg_replace('/[^bkmgtpezy]/i', '', $size);
-	// Remove the non-numeric characters from the size.
-	$size = preg_replace('/[^0-9\.]/', '', $size);
-	if ($unit)
-	{
-		// Find the position of the unit in the ordered string which is the power
-		// of magnitude to multiply a kilobyte by.
-		return round($size * pow(1024, stripos('bkmgtpezy', $unit[0])));
-	}
-	else {
-		return round($size);
-	}
-}
-
-
 //- Check user authentication, login and logout
 $auth = new Authorization(); //create authorization object
 
 // check if user has attempted to log out
-if (isset($_GET['logout']))
+if (isset($_POST['logout']))

```

Comment:
But ... why? It really doesn't matter if this is a GET or a POST request. You could make the argument that it changes state on the server so it _should_ be a POST request, but that's hardly ever enforced for session state type changes that only affect to a logout state. For a login state, sure, but logout?